contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -54,20 +53,19 @@ contract DisputeManagerFacet is StateChannelCommon {
         DisputeAuditingData memory disputeAudit
     ) public returns (bool isSuccess, address[] memory slashParticipants) {
 
-        address challenger = msg.sender;
         // check if the commitment of dispute is available
         bytes32 memory disputeCommitment = keccak256(abi.encodePacked(
             dispute,
             block.timestamp
         ));
         (bool isAvailable, int index) = isDisputeCommitmentAvailable(disputeCommitment);
         if(!isAvailable) {
-            revert("Dispute commitment not available");
+            revert(abi.encode("AUDIT: DISPUTE COMMITMENT NOT AVAILABLE"));
```

Comment:
This should be revert(string) not revert(abi.encode(string)) - The string is implicitly interpreted by ethers and all/most EVM tooling, but this requires more gas and larger contract size -> better to use custom Errors

Reply:
now i ma first returning the bytes as values and not reverting, if thats make sense
so that the client can inteprete it based on boolean value.

"The main auditing logic on success returns true with the list of slashed participants (useful for [onChainSlashes](https://docs.google.com/document/d/1h1MmOhTgVFVbf3BYc0Di9RdWA-PS4tEGARVOspQv2kE/edit?tab=t.0#bookmark=id.d8irzt63u2xe)) and on failure returns false with an error reason (out of gas, missing oracle, etc..). (Hint: for solidity use bytes and interpret them based on the boolean value)"

Reply:
What if I return an enum of ErrorType instead of strings So that the function signature to be consistent

returns (bool success, address[], EnumError), hence when handling we just check the success and process

Reply:
The thing is some errors will have to have additional information, for example "blockcalldata commitment is missing for [channelId] [senderAddress] [fork] [blockNumber]"

We should also have an error "AuditingData incorrect" - since the auditor provides the data, so if the auditor provides junk, auditing will fail, but we shouldn't slash the backer for it, but the auditor - even though auditing failed. 

If the backer commits to junk, the auditor should be able to prove it with a separate fraud proof.

The error is mostly useful to the off-chain peer to try and fetch the commitment if its off-chain environment isn't synced with the on-chain one + in the case of wrong 'AuditingData' it's useful on-chain too.